### PR TITLE
Feat chat

### PR DESCRIPTION
**Pull request by capossele**

### CODE REVIEW

File: docs/tutorials/dApp.md [hunk 1]
```diff
@@ -1,39 +1,157 @@
...
+Throughout this tutorial we will learn how to write simple dApps as GoShimmer plugins. We provide two different examples: a chat dApp and a network delay dApp. Hope you enjoy the reading!
+
+## Chat dApp
+In this guide we are going to explain how to write a very simple chat dApp so that anyone, connected to a GoShimmer node, could write a short message and read what is being written into the Tangle.
+
+The complete source code of the application can be found [in the repository](https://github.com/iotaledger/goshimmer/tree/develop/plugins/chat). 
+
+### Overview
+Our chat dApp can be implemented in a few simple steps:
+1. A node sends via the Tangle a special message containing a chat payload.
```
Comment (jonastheis) [inline docs/tutorials/dApp.md:18]:
```suggestion
1. A node sends a special message containing a chat payload via the Tangle.
```

File: docs/tutorials/dApp.md [hunk 2]
```diff
@@ -1,39 +1,157 @@
...
+2. Upon receipt, every other node in the network process this message and trigger an event that a chat message has beein received.
```
Comment (jonastheis) [inline docs/tutorials/dApp.md:19]:
```suggestion
2. Upon receipt, every other node in the network processes this message and - if the chat dApp/plugin is enabled - triggers an event that a chat message has been received.
```

Comment (jkrvivian) [inline docs/tutorials/dApp.md:19]:
```suggestion
2. Upon receipt, every other node in the network process this message and trigger an event that a chat message has been received.
```

File: docs/tutorials/dApp.md [hunk 3]
```diff
@@ -1,39 +1,157 @@
...
+	}
+	if consumedBytes != len(data) {
+		return nil, errors.New("not all payload bytes were consumed")
+	}
+	return
+})
+```
+
+### Create The Web API Endpoints
+In order to issue a message with our newly created chat payload, we need to create a web API endpoint. Here we simply bind a json request containing the necessary fields: `from`, `to` and `message` and then issue it into the Tangle with `messagelayer.Tangle().IssuePayload(chatPayload)`. This plugin takes care of all the specifics and employs the `MessageFactory` to, i.a., select tips and sign the message.
```
Comment (jkrvivian) [inline docs/tutorials/dApp.md:69]:
```suggestion
In order to issue a message with our newly created chat payload, we need to create a web API endpoint. Here we simply bind a json request containing the necessary fields: `from`, `to` and `message` and then issue it into the Tangle with `messagelayer.Tangle().IssuePayload(chatPayload)`. This plugin takes care of all the specifics and employs the `MessageFactory` to, i.e., select tips and sign the message.
```

File: plugins/chat/plugin.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/iotaledger/goshimmer/plugins/messagelayer"
+)
+
+const (
+	// PluginName contains the human readable name of the plugin.
+	PluginName = "Chat"
+)
+
+var (
+	// App is the "plugin" instance of the network delay application.
```
Comment (jonastheis) [inline plugins/chat/plugin.go:19]:
```suggestion
	// App is the "plugin" instance of the chat application.
```

**Review by jonastheis — APPROVED** [review]

File: plugins/dashboard/chat_livefeed.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		}})
+
+		task.Return(nil)
+	}, workerpool.WorkerCount(chatLiveFeedWorkerCount), workerpool.QueueSize(chatLiveFeedWorkerQueueSize))
+}
+
+func runChatLiveFeed() {
+	if err := daemon.BackgroundWorker("Dashboard[ChatUpdater]", func(shutdownSignal <-chan struct{}) {
+		newMsgRateLimiter := time.NewTicker(rate)
+		defer newMsgRateLimiter.Stop()
```
Comment (jkrvivian) [inline plugins/dashboard/chat_livefeed.go:51]: Why do we need a ticker?
just send out the chat msg whenever we received one.

File: docs/tutorials/dApp.md (hunk 3 above)
Comment (capossele) [reply docs/tutorials/dApp.md:69]: i.a., is correct in this context

**Pull request merged**